Update conventional-changelog dependency

jeffbcross opened the pull request:
I...
- updated our conventional-changelog version to latest, 
- installed conventional-changelog-cli, 
- deleted our old changelog generation script, 
- and added a new npm script (`npm run gen-changelog`)

I tested that this fixes #5672 by creating this commit:

```
refactor(test): testing changeling generation do not submit

BREAKING CHANGE:

THIS IS BREAKING!!!
```

Then I ran `npm run gen-changelog`

Then I checked CHANGELOG.md, which contained this markdown (and prepended safely without destroying the rest of the changelog).

`<begin CHANGELOG snippet>`
### Code Refactoring
- **test:** testing changeling generation do not submit ([17775b2](https://github.com/angular/angular/commit/17775b2))
### BREAKING CHANGES
- test: THIS IS BREAKING!!!

`<end CHANGELOG snippet>`

TODO: 
- [ ] Update our release docs to reflect new changelog script

jeffbcross committed:
build(npm): update conventional-changelog dependency

Fixes #5672

jeffbcross committed:
build(changelog): remove old changelog script and add gulp task

IgorMinar commented on package.json:
```diff
@@ -15,7 +15,8 @@
   "scripts": {
-    "postinstall": "node tools/npm/copy-npm-shrinkwrap && webdriver-manager update"
+    "postinstall": "node tools/npm/copy-npm-shrinkwrap && webdriver-manager update",
+    "gen-changelog": "conventional-changelog -p angular -i CHANGELOG.md -s -r 1"
```
can you please move this to gulp instead. in gulp, package.json and bash scripts :)

jeffbcross replied:
Sure, just a task called `changelog`?

IgorMinar commented:
the rest looks great. thank you

jeffbcross commented:
I updated the PR...
 * removed the conventional-changelog-cli dependency
 * removed `gen-changelog` npm script
 * added gulp-conventional-changelog dependency
 * added gulp task `changelog`
 * verified `CHANGELOG.md` output was same as previous test

IgorMinar commented:
lgtm

jeffbcross commented:
Merged at 0ad1215a92e16aa36f3969e6116f68f4aa854cba
